Frameworks/Foundation/NSString.mm

```diff
@@ -680,7 +680,25 @@ - (instancetype)pathExtension {
  @Status Interoperable
 */
 - (instancetype)lastPathComponent {
-    return [self substringFromIndex:_CFStartOfLastPathComponent2(static_cast<CFStringRef>(self))];
+    if ([self length] == 0) {
+        return @"";
+    }
+
+    CFIndex index = _CFStartOfLastPathComponent2(static_cast<CFStringRef>(self));
+    NSString* substring = [self substringFromIndex:index];
+
+    if (IS_SLASH([self characterAtIndex:index])) {
+        if ([self length] == 1) {
+            return substring;
+        }
+
+        // only way we could of ended up here is, if we have a string with multiple leading slashes.
+        NSArray<NSString*>* tokens = [self componentsSeparatedByString:_NSGetSlashStr()];
+        return ([tokens[0] isEqualToString:@""]) ? _NSGetSlashStr() : tokens[0];
+    }
+
+    NSArray<NSString*>* tokens = [substring componentsSeparatedByString:_NSGetSlashStr()];
+    return tokens[0];
```

Comment:
Okay - please add a comment that this step removes the trailing slashes still in the substring, with the example you just gave me :D

---
In reply to: [74354542](https://github.com/Microsoft/WinObjC/pull/870#discussion_r74354542) [](ancestors = 74354542,74354418,74354338,74353642)